STUDIO/org.openl.rules.webstudio/src/org/openl/rules/rest/settings/service/impl/AuthenticationSettingsFactoryImpl.java

```diff
@@ -0,0 +1,40 @@
+package org.openl.rules.rest.settings.service.impl;
+
+import org.openl.config.InMemoryProperties;
+import org.openl.rules.rest.settings.service.AuthenticationSettingsFactory;
+import org.openl.rules.webstudio.web.admin.security.ADAuthenticationSettings;
+import org.openl.rules.webstudio.web.admin.security.AuthenticationSettings;
+import org.openl.rules.webstudio.web.admin.security.CASAuthenticationSettings;
+import org.openl.rules.webstudio.web.admin.security.InheritedAuthenticationSettings;
+import org.openl.rules.webstudio.web.admin.security.NOPAuthenticationSettings;
+import org.openl.rules.webstudio.web.admin.security.OAuth2AuthenticationSettings;
+import org.openl.rules.webstudio.web.admin.security.SAMLAuthenticationSettings;
+import org.openl.rules.webstudio.web.admin.security.UserMode;
+
+public class AuthenticationSettingsFactoryImpl implements AuthenticationSettingsFactory {
+
+    private final InMemoryProperties properties;
+
+    public AuthenticationSettingsFactoryImpl(InMemoryProperties properties) {
+        this.properties = properties;
+    }
+
+    @Override
+    public AuthenticationSettings initialize() {
+        var userMode = UserMode.fromValue(properties.getProperty(AuthenticationSettings.USER_MODE));
+        return create(userMode);
+    }
+
+    @Override
+    public AuthenticationSettings create(UserMode userMode) {
+        return switch (userMode) {
+            case DEMO -> new AuthenticationSettings();
```

Comment:
Remove DEMO and CAS, they will not supported on the new UI

Reply:
done